dash/_callback.py

```diff
@@ -462,7 +473,15 @@ def add_context(*args, **kwargs):
                 if output_value is callback_manager.UNDEFINED:
                     return to_json(response)
             else:
-                output_value = _invoke_callback(func, *func_args, **func_kwargs)
+                try:
+                    output_value = _invoke_callback(func, *func_args, **func_kwargs)
+                except Exception as err:  # pylint: disable=broad-exception-caught
+                    if on_error:
```

Comment:
Why not do something like `on_error_func = on_error or app_on_error` further up in the code, to avoid this extra `elif`